### ISSUE TITLE
An 500 error when publishing a specific event

### ISSUE THREAD

**Issue by adulau** [MISP/MISP#560]
We have a strange bug (that seems to come from the XML library in CakePHP). This only happens with some specific recent events. 

~~~~
2015-07-08 06:35:51 Warning: Warning (2): SimpleXMLElement::__construct(): Entity: line 170: parser error : internal error: Huge input lookup in [XXX/app/Lib/cakephp/lib/Cake/Utility/Xml.php, line 221]
Trace:
SimpleXMLElement::__construct() - [internal], line ??
Xml::fromArray() - APP/Lib/cakephp/lib/Cake/Utility/Xml.php, line 221
Event::restfullEventToServer() - APP/Model/Event.php, line 619
Event::uploadEventToServer() - APP/Model/Event.php, line 496
Event::uploadEventToServersRouter() - APP/Model/Event.php, line 1457
Event::publish() - APP/Model/Event.php, line 1530
Event::publishRouter() - APP/Model/Event.php, line 1497
EventsController::publish() - APP/Controller/EventsController.php, line 1478
ReflectionMethod::invokeArgs() - [internal], line ??
Controller::invokeAction() - APP/Lib/cakephp/lib/Cake/Controller/Controller.php, line 490
Dispatcher::_invoke() - APP/Lib/cakephp/lib/Cake/Routing/Dispatcher.php, line 185
Dispatcher::dispatch() - APP/Lib/cakephp/lib/Cake/Routing/Dispatcher.php, line 160
[main] - APP/webroot/index.php, line 92

~~~~

**Comment by iglocska** [comment]
Wow, never seen that one before! Looking into it.

**Comment by elhoim** [comment]
PHP memory consumption limit?
Or same thing but for the XML parsing library?

A quick googling:
http://stackoverflow.com/questions/14950589/simplexml-load-string-errors-on-big-files-occur-on-one-system-but-not-another

**Comment by iglocska** [comment]
Think it's the latter that's what I found too. I'll look into switching it to my own xml builder, it's much more simplistic than simplexml as we don't need a lot of the functionality.

**Comment by adulau** [comment]
The XML is indeed 12MB but there are other bigger event where the issue doesn't occur. It's event 1527 in CIRCL private sector MISP for the ones having access.

**Comment by iglocska** [comment]
It probably has to do with a single string being too large for it, also
simplexml is utterly horrible with memory usage (the conversion will take
up about 10x as much memory as the actual data). I am moving it over to my
own converter, surprised that I haven't done it yet for that specific
method (all the xml views already use it).

On Wed, Jul 8, 2015 at 9:06 AM, Alexandre Dulaunoy <notifications@github.com

**Comment by iglocska** [comment]
Fixed in 6a25471ea435149236ae502bed83fd16a6040214

**Comment by adulau** [comment]
Tested. It's fixed. Thanks a lot for the prompt and clean fix.
